### PR TITLE
Pal 1052 fix audit kafka tests

### PR DESCRIPTION
**Pull request by pd104923**

### CODE REVIEW

File: audit-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/audit/kafka/KafkaContractTest.java [hunk 1]
```diff
@@ -190,11 +192,22 @@ void testFailedErrorDeserialisation() throws Exception {
...
+                        .as("Check exactly 1 'Error' file has been created")
+                        .isEqualTo(expectedErrorCount);
+            } catch (AssertionError e) {
+                LOGGER.info("There was an assertion error on attempt {}", i);
+                if (i == N_RUNS) {
+                    throw e;
+                }
+            }
+            TimeUnit.SECONDS.sleep(BACKOFF);
+        }
```
Comment (dev930018) [inline audit-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/audit/kafka/KafkaContractTest.java]: Can we stick this in a `tryAssertWithBackoff` method or something?
Pass a `Runnable` as an argument, which here would be `() -> assertThat(currentErrorCount.get()).as(...).isEqualTo(...)`
Use it in the method as `try { runnable.run() } catch (AssertionError e) { ... }`

Comment (dev930018) [reply audit-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/audit/kafka/KafkaContractTest.java]: Then if we want more tests here, or want to re-use this piece of code, it's straightforward and we're not repeating ourselves too much

**Review by dev930018 — CHANGES_REQUESTED** [review]

File: audit-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/audit/kafka/KafkaContractTest.java [hunk 2]
```diff
@@ -239,4 +246,15 @@ private void waitForService() throws InterruptedException {
...
+    private void tryAssertWithBackoff(final Runnable runnable, final int i) throws InterruptedException {
+        try {
+            runnable.run();
+        } catch (AssertionError e) {
+            LOGGER.info("There was an assertion error on attempt {}", i);
+            if (i == N_RUNS) {
+                throw e;
+            }
+        }
+        TimeUnit.SECONDS.sleep(BACKOFF);
```
Comment (dev930018) [inline audit-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/audit/kafka/KafkaContractTest.java]: Was expecting this to have both the `try` and the `for`

**Review by dev930018 — CHANGES_REQUESTED** [review]

File: audit-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/audit/kafka/KafkaContractTest.java (hunk 2 above)
Comment (pd104923) [reply audit-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/audit/kafka/KafkaContractTest.java]: Done

**Review by dev930018 — DISMISSED** [review]

File: audit-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/audit/kafka/KafkaContractTest.java [hunk 3]
```diff
@@ -212,23 +214,22 @@ void testFailedSuccessDeserialisation() throws Exception {
 
         // Then check a "Success-..." file has been created
         var actualSuccessCount = currentSuccessCount.get();
-        assertThat(actualSuccessCount)
+        tryAssertWithBackoff(() ->assertThat(actualSuccessCount)
```
Comment (ac74475) [inline audit-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/audit/kafka/KafkaContractTest.java]: fix checkstyle issue of space after ->

**Review by dev930018 — APPROVED** [review]

**Pull request merged**